src/facade/dragonfly_connection.cc

```diff
@@ -300,22 +302,24 @@ void Connection::HandleRequests() {
 
   auto remote_ep = lsb->RemoteEndpoint();
 
+  FiberSocketBase* peer = socket_.get();
 #ifdef DFLY_USE_SSL
   unique_ptr<tls::TlsSocket> tls_sock;
   if (ctx_) {
-    tls_sock.reset(new tls::TlsSocket(socket_.get()));
-    tls_sock->InitSSL(ctx_);
-
-    FiberSocketBase::AcceptResult aresult = tls_sock->Accept();
-    if (!aresult) {
-      LOG(WARNING) << "Error handshaking " << aresult.error().message();
-      return;
+    const bool no_tls_on_admin_port = absl::GetFlag(FLAGS_no_tls_on_admin_port);
+    if (!no_tls_on_admin_port || (no_tls_on_admin_port && !IsAdmin())) {
```

Comment:
I find this a bit more readable -
```suggestion
    if (!(IsAdmin() && no_tls_on_admin_port)) {
```